hub/api/dataset_utils.py

```diff
@@ -51,3 +54,20 @@ def slice_extract_info(slice_, num):
     elif slice_.start is not None and slice_.stop is None:
         num = num - slice_.start
     return num, offset
+
+
+def str_to_int(assign_value, tokenizer):
+    assign_value = assign_value.decode("utf-8") if isinstance(assign_value, bytes) else assign_value
+    if (isinstance(assign_value, np.ndarray) and assign_value.dtype.type is np.bytes_) or (isinstance(assign_value, list) and isinstance(assign_value[0], bytes)):
+        assign_value = [item.decode("utf-8") for item in assign_value]
+    if tokenizer is not None:
+        from transformers import AutoTokenizer
```

Comment:
if transformers is not imported, you should through a ModuleNotFounder Exception, check how TensorFlow or PyTorch are loaded.